File: array-1637.py
points = [[3,1],[9,0],[1,0],[1,4],[5,3],[8,8]]

x = []

for point in points:
   x.append(point[0])

x.sort()

mx = 0
for i in range(1, len(x)):
    mx = max(mx,  x[i] - x[i-1])

print(mx)

File: array-1637.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main(){
    std::vector<std::vector<int>> points = {{8,7},{9,9},{7,4},{9,7}};
    std::vector<int> x;
    int count = 0;

    for(int i = 0; i<points.size(); i++){
            x.push_back(points[i][0]);
    }

    std::sort(x.begin(), x.end());

    int ans = 0;

    for (int i = 1; i < x.size(); i++) {
        ans = std::max(ans, x[i] - x[i - 1]);
    }

    std::cout << ans;
}